src/firewall/core/nftables.py

```diff
@@ -1594,15 +1594,12 @@ def _rich_rule_limit_fragment(self, limit):
             "d": "day",
         }
 
-        try:
-            i = limit.value.index("/")
-        except ValueError:
-            raise FirewallError(INVALID_RULE, "Expected '/' in limit")
+        rate, duration = limit.value_parse()
```

Comment:
This usage makes me think `value_parse()` is the wrong name for the function. If I read your commit message right the value is already parsed and normalized.

These make more sense to me:
```
rate, duration = tuple(limit)
```
or
```
rate, duration = limit.tuple()
````

Then later when you add burst the tuple size increases, but python is smart.
```
>>> foo, bar = (1,2,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```

Individual values are still available via `obj.value` and `obj.burst`.

---

> Now parse and stringify the value, so that it is normalized. Invalid
values are left unnormalized, and Rich_Limit.__init__() still does not
fail the operation (like before). For that we have check().

Reply:
>  If I read your commit message right the value is already parsed and normalized.

The values is normalized (if and only if it's normalizable). Normalization requires parsing. But it's still a string, that cannot be understood without parsing it.

some callers want to decompare the value into their parts. They want to parse it. Hence, there is still a parse, even after having the values normalized.

Yes, as the Rich_Rule setter already parses the value to normalize it, it could also keep around the parsed values. Like setting `self.duration` & `self.rate`. However, that means all those object, that just lie around in memory for the most part, also hold the redundant, parsed values. Instead, the parsing result is not kept. You have to re-request it. That is intentional, to avoid storing those values. It could of course be different. It trades some CPU vs. Memory. The usage is pretty much the same.

What a separate parse also gives, is that a Rich_Limit can also hold invalid value, and only blow up during check(). That is simpler, if the parsing is always done anew.

> But that assumes you fully parsed/normalized in the constructor.

The constructor does normalization via `self.value = value`. That implies it also does a parsing. However, it doesn't keep those parse-results around...